jme3-core/src/main/java/com/jme3/anim/AnimLayer.java

```diff
@@ -0,0 +1,225 @@
+    /**
+     * The composer that owns this layer. Were it not for cloning, this field
+     * would be final.
+     */
+    private AnimComposer composer;
+    /**
+     * Limits the portion of the model animated by this layer. If null, this
+     * layer can animate the entire model.
+     */
+    private final AnimationMask mask;
+    /**
+     * The current animation time, in scaled seconds. Always non-negative.
+     */
+    private double time;
+    /**
+     * The software object (such as an AnimEvent) that currently controls this
+     * layer, or null if unknown.
+     */
+    private Object manager;
+
+    /**
+     * Instantiates a layer without a manager or a current Action, owned by the
+     * specified composer.
+     *
+     * @param composer the owner (not null, alias created)
+     * @param mask the AnimationMask (alias created) or null to allow this layer
+     *     to animate the entire model
+     */
+    AnimLayer(AnimComposer composer, AnimationMask mask) {
+        assert composer != null;
+        this.composer = composer;
+
+        this.mask = mask;
+    }
+
+    /**
+     * Returns the Action that's currently running.
+     *
+     * @return the pre-existing instance, or null if none
+     */
+    public Action getCurrentAction() {
+        return currentAction;
+    }
+
+    /**
+     * Returns the current manager.
+     *
+     * @return the pre-existing object (such as an AnimEvent) or null for
+     *     unknown
+     */
+    public Object getManager() {
+        return manager;
+    }
+
+    /**
+     * Returns the animation mask.
+     *
+     * @return the pre-existing instance, or null if this layer can animate the
+     *     entire model
+     */
+    public AnimationMask getMask() {
+        return mask;
+    }
+
+    /**
+     * Returns the animation time, in scaled seconds.
+     *
+     * @return the current animation time (not negative)
+     */
+    public double getTime() {
+        return time;
+    }
+
+    /**
+     * Runs the specified Action, starting from time = 0. This cancels any
+     * Action previously running on this layer.
+     *
+     * @param actionToRun the Action to run (alias created) or null for no
+     *     action
+     */
+    public void setCurrentAction(Action actionToRun) {
+        this.time = 0.0;
+        this.currentAction = actionToRun;
+    }
+
+    /**
+     * Assigns the specified manager. This cancels any manager previously
+     * assigned.
+     *
+     * @param desiredManager the desired manager (such as an AnimEvent, alias
+     *     created) or null for unknown manager
+     */
+    public void setManager(Object desiredManager) {
+        this.manager = desiredManager;
+    }
+
+    /**
+     * Changes the animation time, wrapping the specified time to fit the
+     * current Action. An Action be running.
+     *
+     * @param desiredAnimationTime the desired time (in scaled seconds)
+     */
+    public void setTime(double desiredAnimationTime) {
```

Comment:
Simplify the variable name 'desiredAnimationTime' to 'animTime'. The javadoc already clarifies the meaning of the parameters very well. eg: 
```java
@param animTime the desired time (in scaled seconds)
```

Reply:
ok